kyaml/fn/framework/selector.go

```diff
@@ -200,8 +214,22 @@ func (s AndSelector) Filter(items []*yaml.RNode) ([]*yaml.RNode, error) {
 // DefaultTemplateData makes AndSelector a ResourceTemplateMatcher.
 // Although it does not contain templates itself, this allows it to support ResourceTemplateMatchers
 // when being used as a matcher itself.
-func (s AndSelector) DefaultTemplateData(data interface{}) {
+func (s *AndSelector) DefaultTemplateData(data interface{}) {
 	if s.TemplateData == nil {
 		s.TemplateData = data
 	}
 }
+
+func (s *AndSelector) InitTemplates() error {
```

Comment:
nit: This method and the OrSelector's one are exactly the same, maybe just write a function that both call?